spec/result.json

```diff
@@ -0,0 +1,73 @@
+{
+    "$schema": "http://json-schema.org/draft-07/schema#",
+    "type": "object",
+    "required": [
+        "head"
+    ],
+    "properties": {
+        "head": {
+            "type": "object",
+            "properties": {
+                "additionalProperties": false,
+                "link": {
+                    "type": "array",
+                    "items": {
+                        "type": "string",
+                        "format": "uri"
+                    }
+                },
+                "vars": {
+                    "type": "array",
+                    "items": {
+                        "type": "string"
+                    }
+                }
+            }
+        },
+        "boolean": {
+            "type": "boolean"
+        },
+        "results": {
+            "type": "object",
+            "required": [
+                "bindings"
+            ],
+            "properties": {
+                "bindings": {
+                    "type": "array",
+                    "items": {
+                        "type": "object",
+                        "additionalProperties": {
+                            "type": "object",
+                            "properties": {
+                                "type": {
+                                    "type": "string",
+                                    "enum": [
+                                        "uri",
+                                        "literal",
+                                        "bnode"
+                                    ]
+                                },
+                                "value": {
+                                    "type": "string"
+                                },
+                                "xml:lang": {
+                                    "type": "string",
+                                    "maxLength": 35
+                                },
+                                "datatype": {
```

Comment:
Is it possible to make it so that `xml:lang` and `datatype` are exclusive?

Reply:
I believe it's possible with the `enum` enum. It should be also possible to state that `xml:lang` and `datatype` should only be used when `"type": "literal"` and that `"boolean"` and `"results"` are exclusive (and that `"header"/"vars" must be used only when `"results"` is present.

Reply:
@afs what about `rdf:langString`?